dot/node.go

```diff
@@ -41,9 +48,87 @@ type Node struct {
 	started  chan struct{}
 }
 
+//go:generate mockgen -source=node.go -destination=mock_node_test.go -package=$GOPACKAGE
+
+type newNodeIface interface {
```

Comment:
nice! I like this interface a lot, maybe name it `nodeBuilder` or `nodeCreator` or something like that?